pkgs/tools/compression/zstd/default.nix

```diff
@@ -24,7 +24,13 @@ stdenv.mkDerivation rec {
     # This patches makes sure we do not attempt to use the MD5 implementation
     # of the host platform when running the tests
     ./playtests-darwin.patch
-  ];
+  ] ++ lib.optional stdenv.is32bit
+    (fetchpatch { # https://github.com/facebook/zstd/pull/2606
+      name = "test-memory-usage.diff";
+      url = "https://github.com/facebook/zstd/commit/6f40571a.diff";
```

Comment:
This should be the long rev ``6f40571ae2feb8bfa0a56f9871b6ee3084085fc2`` and also we can probably apply it unconditionally which probably prevents a breakage on the next update cause no one tests i686 tbh.

Reply:
The major update most likely won't be suitable for backporting.  (it's in `staging` ATM)  The output hash is full, so I don't think there's noticeable risk here, but I don't really care.

Reply:
Anyway, we could apply it unconditionally on some mass rebuild, even though the benefits seem low.

Reply:
The problem is if an update causes the patch to not apply most likely no one will notice it.

Also the problem with the short rev is that I could create a commit that begins with 6f40571a which causes the URL to return a 404 and essentially DOSing people who can't cache the file.

Reply:
Yes, neither felt that bad to me: relatively unlikely and easy to fix when they happen.  So let's fix them (as you seem to care more than me); it's mainly that the big rebuild isn't very practical at _this_ moment.

Reply:
Staged as 99995bc9308.